### DISCUSSION TITLE
Can't upgrade from 1.x to 2.x in mixed code project (native code with /clr switch)

### DISCUSSION THREAD

**Discussion by stalek71** [xunit/xunit#2279] [Question] [answered]
When I try to add the latest version of XUnit to a mixed project with clr support (c++/cli managed extensions) it says I can't upgrade it from version 1.x. It's a project having mixed code (native and clr) and I was able to add xunit.dll in version 1.x manually - by pointing it out directly on a disk.

This is a message I receive:
```
Could not install package 'xunit.assert 2.4.1'. You are trying to install this package 
into a project that targets 'native,Version=v0.0', but the package does not contain 
any assembly references or content files that are compatible with that framework. 
For more information, contact the package author.
```

What to do with it?

**Comment by bradwilson** [comment, marked as answer]
I don't know that I can be much help here. Native code has never officially been a supported thing for us.

A couple thoughts:

- Can you try upgrading from 1.9.2 to 2.0.0 and see what happens? At least from 2.0.0 (which was a Profile259 Portable Class Library solution) to 2.4.1 (which uses the newer `netstandard` targets), it would be interesting to know what kind of support might be missing from the native compiler tool chain.
- Which version of Visual Studio are you using?
